platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/deduplication/EventDeduplicator.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.event.deduplication;
+
+import static com.swirlds.common.metrics.Metrics.PLATFORM_CATEGORY;
+
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.metrics.LongAccumulator;
+import com.swirlds.common.sequence.map.SequenceMap;
+import com.swirlds.common.sequence.map.StandardSequenceMap;
+import com.swirlds.common.system.events.EventDescriptor;
+import com.swirlds.platform.event.GossipEvent;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Consumer;
+
+/**
+ * Deduplicates events.
+ * <p>
+ * A duplicate event is defined as an event with an identical descriptor, and identical signature to an event that has
+ * already been observed.
```

Comment:
Nitpick. In general I think a list of two things doesn't deserve a comma in the middle (although technically speaking it's not grammatically incorrect).

```suggestion
 * A duplicate event is defined as an event with an identical descriptor and identical signature to an event that has
 * already been observed.
```